app/services/foreman_bootdisk/iso_generator.rb

```diff
@@ -41,6 +47,15 @@ def self.generate(opts = {}, &block)
       initrd /script
     EOS
 
+    opts[:grub] = <<-EOS if opts[:grub].nil? && opts[:ipxe]
+      set default=0
+      set timeout=5
+      menuentry "Chainload iPXE chain" {
+        search --no-floppy --set=root -f /ipxe.efi
+        chainloader ($root)/ipxe.efi cmdline "echo If patch was applied you should see this text ; sleep 10"
```

Comment:
Change this so it actually works, we need to error out early for static IP allocation and make generic bootdisk working. Pseudo code:

```ruby
if opts[:grub].nil? && opts[:ipxe]
  raise ::Foreman::Exception.new(N_('Cannot generate UEFI iPXE disk for DHCP-less network')) unless host.provision_interface.subnet.dhcp_boot_mode?
  iface = host.host_param("bootdisk_if") || "net0"
  opts[:grub] = <<-EOS
    ...
    chainloader ($root)/ipxe.efi cmdline "dhcp #{iface} & chain #{bootdisk_chain_url}${#{iface}/mac}"
```

You get the idea.